level2/load_balancer.tf

```diff
@@ -1,8 +1,104 @@
-module "lb" {
-  source = "../modules/lb"
+data "aws_route53_zone" "main" {
+  name = "techstellar.link"
+}
+
+module "acm" {
+  source = "terraform-aws-modules/acm/aws"
+
+  domain_name = "www.techstellar.link"
+  zone_id     = data.aws_route53_zone.main.zone_id
+
+  validation_method = "DNS"
+
+  wait_for_validation = true
+}
+
+module "external_sg" {
+  source = "terraform-aws-modules/security-group/aws"
+
+  name   = "${var.env_code}-external"
+  vpc_id = data.terraform_remote_state.level1.outputs.vpc_id
+
+  ingress_with_cidr_blocks = [
+    {
+      from_port   = 443
+      to_port     = 443
+      protocol    = "tcp"
+      description = "https to ELB"
+      cidr_blocks = "0.0.0.0/0"
+    }
+  ]
+
+
+  egress_with_cidr_blocks = [
+    {
+      from_port   = 0
+      to_port     = 65535
+      protocol    = "tcp"
+      description = "https to ELB"
+      cidr_blocks = "0.0.0.0/0"
+    }
+  ]
+
+}
+
+module "elb" {
+  source  = "terraform-aws-modules/alb/aws"
+  version = "8.4.0"
+  name    = var.env_code
+
+  load_balancer_type = "application"
+
+  vpc_id          = data.terraform_remote_state.level1.outputs.vpc_id
+  internal        = false
+  subnets         = data.terraform_remote_state.level1.outputs.public_subnet_id
+  security_groups = [module.external_sg.security_group_id]
+
+  target_groups = [
+    {
+      name_prefix          = var.env_code
+      backend_protocol     = "HTTP"
+      backend_port         = 80
+      deregistration_delay = 10
+
+      health_check = {
+        enabled             = true
+        internal            = 30
+        path                = "/"
+        port                = "traffic-port"
+        healthy_threshold   = 5
+        unhealthy_threshold = 2
+        timeout             = 5
+        protocol            = "HTTP"
+        matcher             = "200"
+      }
+    }
+  ]
+  target_group_arns = module.elb.target_group_arns
```

Comment:
wrong. you are meant to define this in the asg module.

Reply:
Resolved and committed